src/main/scala/scalafix/sbt/ScalafixPlugin.scala

```diff
@@ -305,7 +341,16 @@ object ScalafixPlugin extends AutoPlugin {
     val parser = Def.setting(
       new ScalafixCompletions(
         workingDirectory = (ThisBuild / baseDirectory).value.toPath,
-        loadedRules = () => scalafixInterfaceProvider.value().availableRules(),
+        loadedRules = () =>
+          // since the introduce of scalafixSbtResolversAsCoursierRepositories
+          // loading rules here without converted sbt resolvers could cause trying to download dependencies that unable to resolve
+          // we simply recover it to empty external rules to maintain the main task runs well
```

Comment:
We actually won't show any rule (even the built-in ones) but that's OK. I find this comment a bit hard to grasp. Here is my suggestion
```suggestion
          // `scalafixDependencies` might be resolvable only from one of the
          // `scalafixSbtResolversAsCoursierRepositories` that we can't look up
          // here, as it's a task and we are in a settings context. Therefore
          // we fallback on an empty list of rules in case of resolution error
          // to keep providing completions for the rest.
```
Also, considering the amount of duplicated code/comment, I think it's time to extract to some `private def parser: Def.Initialize[Parser[ShellArgs]]`

Reply:
extracted into 'scalafixParser'

Reply:
there was actually one difference between scalafix and scalafixAll, which the scripted test guarded, and which I have reintroduced via 2ddd9cc5e8ace38053d371274cb43e05aafe0804